### PR TITLE
Fixes check style warnings and errors.

### PR DESCRIPTION
**Pull request by eddieajau**

### CODE REVIEW

**Comment by ianmacl** [comment]
# Test Results

Total Tests: 1420
Assertions: 9898
Failures: 0
Errors: 0
Test Time: 177.701600

File: libraries/joomla/base/node.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ class JNode extends JObject
 	 *
 	 * If the child already has a parent, the link is unset
 	 *
-	 * @param   JNode  $child  The child to be added
```
Comment (ianmacl) [inline libraries/joomla/base/node.php:52]: What does our standard say about object/Class Name?  In file above, we have object, here we have JNode?

**Comment by eddieajau** [comment]
The standard doesn't say anything about it (yet), but I'll add that we should aim to include the correct object type, or at least a common ancestor.  This would be preferred to "object", but I was more worried about eliminating check style warnings on this one.

**Comment by elinw** [comment]
What we have kind of been doing is for @return putting the name in some places ... but it is more readable if we put  

@param object $child A JNode object to be added as the child node.
beause some of those object names are very long.. Sooner or later we will have to do a sweep to make this consistent.

**Comment by eddieajau** [comment]
If you do

@param JNode $child Description

IDE's like PDT will give you type hinting and auto-completion on $child because it knows what class it is.

**Pull request merged**